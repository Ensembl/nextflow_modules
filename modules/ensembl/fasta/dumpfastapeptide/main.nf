// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process FASTA_DUMPFASTAPEPTIDE {
    tag "${db.species}"
    label 'process_low'
    maxForks "${params.max_database_forks}"

    conda "${moduleDir}/environment.yml"
    container 'ensemblorg/ensembl-legacy-scripts:e112_APIv0.4'

    input:
        val(db)

    output:
        tuple val(db), val("fasta_pep"), path("*.fasta"), optional:false, emit: peptide_fasta
        path "versions.yml", emit: versions

    when:
        task.ext.when == null || task.ext.when

    script:
        def args = task.ext.args ?: ''
        def prefix = task.ext.prefix ?: "${db.server.database}"
        def version = "0.4" // No way to get the version from installed repos
        password_arg = db.server.password ? "--pass ${db.server.password}" : ""
        output = "${db.species}_fasta_pep.fasta"

        """
        dump_fasta_peptide.pl \
            --host ${db.server.host} \
            --port ${db.server.port} \
            --user ${db.server.user} \
            ${password_arg} \
            --dbname ${db.server.database} \
            > ${output}

        echo -e -n "${task.process}:\n\tensembl-genomio: ${version}" > versions.yml
        """

    stub:
        def version = "0.4"
        output_file = "pep.fasta"
        dump_dir = "${workflow.projectDir}/tests/modules/ensembl/fasta/dump_peptide/"
        dump_file = "dumped_pep.fasta"
        """
        cp ${dump_dir}/${dump_file} ${output_file}
        echo -e -n "${task.process}:\n\tensembl-genomio: ${version}" > versions.yml
        """
}
