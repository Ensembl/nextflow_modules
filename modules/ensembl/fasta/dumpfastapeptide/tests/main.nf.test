// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test fasta/dumpfastapep
nextflow_process {

    name "Test Process FASTA_DUMPFASTAPEPTIDE"
    script "../main.nf"
    process "FASTA_DUMPFASTAPEPTIDE"

    tag "modules"
    tag "modules_local"
    tag "fasta"
    tag "fasta/dumpfastapeptide"

    test("Fasta Dump Peptide [stub]") {
        when {

                options "-stub"

                process {
                    """
                    input[0] = [[ test_id:'dummy_core_db_core_110_1', species:'Genus species']]
                    """
                }
            }

            then {

                assert snapshot(
                    process.out.peptide_fasta.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("peptide_fasta")

                def testData = process.out.peptide_fasta.collect { core_db, pipeline_tag, fastaPath -> [[ core_db, pipeline_tag, path(fastaPath).fasta]] }
                assert testData.contains([
                    [
                        [['test_id':'dummy_core_db_core_110_1', 'species':'Genus species']], 
                            'fasta_pep', [
                                'PEP2':'MKFLVNVALVFMVVYISYIYAAPEPEPAPEPEAEADAEADPEAGIGAVLK',
                                'PEP1':'APLCOACOIJSOOFAAPOAPKAEWCWCMWCMPWOPSKQPSKQPKSPQKSQ',
                                'PEP3':'AMCAMPACPAPCMPAMPADAPCMPAMCPMAPCPMAPPAMCPMAPMCPAMP'
                            ]
                        ]
                    ]
                )

                assertAll(
                    { assert process.success },
                    { assert snapshot(process.out).match() },
                    { assert process.out.peptide_fasta.size() == 1 },
                    { assert process.out.peptide_fasta != null },
                )
            }
    }
}