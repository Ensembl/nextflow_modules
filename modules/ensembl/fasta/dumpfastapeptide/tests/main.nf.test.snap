{
    "peptide_fasta": {
        "content": [
            [
                [
                    [
                        {
                            "test_id": "dummy_core_db_core_110_1",
                            "species": "Genus species"
                        }
                    ],
                    "fasta_pep",
                    "pep.fasta"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-02T18:03:42.927109271"
    },
    "Fasta Dump Peptide [stub]": {
        "content": [
            {
                "0": [
                    [
                        [
                            {
                                "test_id": "dummy_core_db_core_110_1",
                                "species": "Genus species"
                            }
                        ],
                        "fasta_pep",
                        "pep.fasta:md5,51b32025c7cb77ad3437b6ac129dd0df"
                    ]
                ],
                "1": [
                    "versions.yml:md5,006c65880c710efaa9b684bcb4d6a7a8"
                ],
                "peptide_fasta": [
                    [
                        [
                            {
                                "test_id": "dummy_core_db_core_110_1",
                                "species": "Genus species"
                            }
                        ],
                        "fasta_pep",
                        "pep.fasta:md5,51b32025c7cb77ad3437b6ac129dd0df"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,006c65880c710efaa9b684bcb4d6a7a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-11T14:13:33.965197459"
    }
}